java/org/contikios/cooja/plugins/BufferListener.java

```diff
@@ -699,7 +699,7 @@ public PointerMemoryMonitor(
 
     private void registerSegmentMonitor(int size, boolean notify) throws Exception {
       byte[] pointerValue = mote.getMemory().getMemorySegment(pointerAddress, pointerSize);
-      int segmentAddress = MemoryBuffer.wrap(mote.getMemory().getLayout(), pointerValue).getInt();
+      long segmentAddress = MemoryBuffer.wrap(mote.getMemory().getLayout(), pointerValue).getAddr();
```

Comment:
Should this not read the content of the variable and use as an address and not the address of the variable?

Reply:
I think that is what happens in getAddr(), it does a switch on the size of an address and calls bbuf.getInt() or bbuf.getLong() depending on the architecture.

Reply:
You are right. I did not look at the right code. Sorry.